### ISSUE TITLE
[B2.1] Add backward() method to Variable for simpler API

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#2301]
## Overview
Add a backward() method directly to Variable for a simpler API, instead of requiring manual tape.backward() calls.

## Location
- `shared/autograd/variable.mojo`

## Current State
```mojo
# Current (verbose)
var tape = GradientTape()
var loss = forward_pass(tape)
tape.backward(loss)
```

## Proposed API
```mojo
# Proposed (simpler)
var loss = forward_pass()
loss.backward()  # Automatically triggers backprop
```

## Deliverables
- [ ] Add backward() method to Variable struct
- [ ] Integrate with GradientTape system
- [ ] Handle tape reference appropriately
- [ ] Ensure gradients flow correctly
- [ ] Update docstrings with usage examples

## Success Criteria
- backward() method works on loss Variables
- Gradients computed correctly
- Simpler API for users
- CI passes

## Dependencies
- None (parallel with B2.2)

## Track
Track B: Missing Features (B2.1 of 8)
